llama_index/llms/openai_like.py

```diff
@@ -0,0 +1,66 @@
+from typing import Any, List, Optional
+
+from llama_index.bridge.pydantic import Field
+from llama_index.constants import DEFAULT_CONTEXT_WINDOW
+from llama_index.llms import OpenAI
+
+
+class _Tokenizer(Protocol):
```

Comment:
Nice @benbot!  Thanks!

Seeing this now, I realize I slightly mislead you on this being typed as private because:
- It needs to be defined in `openai` so `OpenAI` can use it to type hint `_get_encoding_for_model`
- It needs to be public so that `openai_like` can import and use it